src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -1,207 +1,375 @@
+                    catch (Exception ex)
+                    {
+                        // Log another exception
+                        _logger.LogError($"Exception has occurred during batch listening:: {ex.ToString()}");
+                    }
+                }
+            }, cancellationToken);
+        }
+
+        private void ThrowIfDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(null);
+            }
+        }
+
         public IScaleMonitor GetMonitor()
         {
             return _scaleMonitor.Value;
-        }
-    }
-}
+        }
+
+        private TriggeredFunctionData GetTriggerFunctionData(Message message, ServiceBusTriggerInput input)
+        {
+            Guid? parentId = ServiceBusCausalityHelper.GetOwner(message);
+            return new TriggeredFunctionData()
+            {
+                ParentId = parentId,
+                TriggerValue = input,
+                TriggerDetails = new Dictionary<string, string>()
+                {
+                    { "MessageId", message.MessageId },
+                    { "DeliveryCount", message.SystemProperties.DeliveryCount.ToString() },
+                    { "EnqueuedTime", message.SystemProperties.EnqueuedTimeUtc.ToString() },
+                    { "LockedUntil", message.SystemProperties.LockedUntilUtc.ToString() },
+                    { "SessionId", message.SessionId }
+                }
+            };
+        }
+
+        private TriggeredFunctionData GetTriggerFunctionData(Message[] messages, ServiceBusTriggerInput input)
+        {
+            Guid? parentId = ServiceBusCausalityHelper.GetOwner(messages[0]);
+
+            int length = messages.Length;
+            string[] messageIds = new string[length];
+            int[] deliveryCounts = new int[length];
+            DateTime[] enqueuedTimes = new DateTime[length];
+            DateTime[] lockedUntils = new DateTime[length];
+            string[] sessionIds = new string[length];
+            for (int i = 0; i < messages.Length; i++)
+            {
+                messageIds[i] = messages[i].MessageId;
+                deliveryCounts[i] = messages[i].SystemProperties.DeliveryCount;
+                enqueuedTimes[i] = messages[i].SystemProperties.EnqueuedTimeUtc;
+                lockedUntils[i] = messages[i].SystemProperties.LockedUntilUtc;
+                sessionIds[i] = messages[i].SessionId;
+            }
+
+            return new TriggeredFunctionData()
+            {
+                ParentId = parentId,
+                TriggerValue = input,
+                TriggerDetails = new Dictionary<string, string>()
+                {
+                    { "MessageIds", string.Join(",", messageIds)},
+                    { "DeliveryCounts", string.Join(",", deliveryCounts) },
+                    { "EnqueuedTimes", string.Join(",", enqueuedTimes) },
+                    { "LockedUntils", string.Join(",", lockedUntils) },

```

Comment:
to avoid the issue with strange names, recommend you just append "Array" to the end of each, as we did in EventGrid binding data for batches [here](https://github.com/Azure/azure-functions-eventhubs-extension/blob/7ccd930e6d2fdde64ded0ec7f540b08627739619/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubTriggerInputBindingStrategy.cs#L102)